applications/mne_analyze/libs/anShared/Model/annotationmodel.cpp

```diff
@@ -594,3 +606,141 @@ MatrixXi AnnotationModel::getAnnotationMatrix()
 
     return matEventDataMatrix;
 }
+
+//=============================================================================================================
+
+int AnnotationModel::createGroup(QString sGroupName, bool bIsUserMade, int iType)
+{
+    EventGroup* newEvent = new EventGroup();
+    int iSize = m_mAnnotationHub.size();
+
+    *newEvent = {iSize,                         //groupNumber
+                 iType,                         //groupType
+                 sGroupName,                    //groupName
+                 bIsUserMade,                   //isUserMade
+                 QVector<int>(),                //dataSamples
+                 QVector<int>(),                //dataTypes
+                 QVector<int>(),                //dataIsUserEvent
+                 QVector<int>(),                //dataSamples_Filtered
+                 QVector<int>(),                //dataTypes_Filtered
+                 QVector<int>()};               //dataIsUserEvent_Filtered
+
+    m_mAnnotationHub.insert(iSize, newEvent);
+
+    return iSize;
+}
+
+//=============================================================================================================
+
+void AnnotationModel::switchGroup(int iGroupIndex)
+{
+    beginResetModel();
+
+    if ((!m_dataSamples.isEmpty()) && (m_iSelectedGroup != ALLGROUPS)){
+        m_mAnnotationHub[m_iSelectedGroup]->dataSamples = m_dataSamples;
+        m_mAnnotationHub[m_iSelectedGroup]->dataTypes = m_dataTypes;
+        m_mAnnotationHub[m_iSelectedGroup]->dataIsUserEvent = m_dataIsUserEvent;
+
+        m_mAnnotationHub[m_iSelectedGroup]->dataSamples_Filtered = m_dataSamples_Filtered;
+        m_mAnnotationHub[m_iSelectedGroup]->dataTypes_Filtered = m_dataTypes_Filtered;
+        m_mAnnotationHub[m_iSelectedGroup]->dataIsUserEvent_Filtered = m_dataIsUserEvent_Filtered;
+    }
+
+    m_dataSamples = m_mAnnotationHub[iGroupIndex]->dataSamples;
+    m_dataTypes = m_mAnnotationHub[iGroupIndex]->dataTypes;
+    m_dataIsUserEvent = m_mAnnotationHub[iGroupIndex]->dataIsUserEvent;
+
+    m_dataSamples_Filtered = m_mAnnotationHub[iGroupIndex]->dataSamples_Filtered;
```

Comment:
Should be `m_dataSamplesFiltered `. Same for `m_dataTypes_Filtered` and `m_dataIsUserEvent_Filtered`